ldms/src/core/ldms_xprt.c

```diff
@@ -3278,7 +3278,7 @@ int ldms_register_notify_cb(ldms_t x, ldms_set_t s, int flags,
 void ldms_xprt_set_delete(ldms_set_t s, ldms_set_delete_cb_t cb_fn, void *cb_arg)
 {
 	struct ldms_request *req;
-	struct ldms_rbuf_desc *rbd, *next_rbd;
+	struct ldms_rbuf_desc *rbd, *next_rbd = NULL;
```

Comment:
@narategithub It appears next_rbd might also be left pointing at something else inappropriate (but not null) from the previous while(rbd) loop when the goto next_1 occurs.
Does this init need to be moved to inside the top of the complaining loop?

Reply:
Nah. If the statements in the first loop got executed, `next_rbd` must be NULL, otherwise, `rbd` won't be NULL and we won't exit the loop.

Thanks for asking though. I'd just saw that the 2nd loop is incorrect. If the first `rbd->xprt` happened to be NULL, then `rbd = next_rbd` will be NULL at `next_1`. I'll add this in another commit on top of this.

Reply:
@baallan please see #619